### ISSUE TITLE
@wordpress/scripts doesn't detect `render` PHP files without a restart

### ISSUE THREAD

**Issue by aurooba** [WordPress/gutenberg#49790]
### Description

If you have `@wordpress/scripts` running with the `start` script, and add a `"render": "file:./render.php"` to your `block.json` file, the script doesn't detect it. But if you end the process and restart it, then it does.

It's very possible I'm just missing something, but I can't figure out why this is happening! I don't recall it being an issue before.

### Step-by-step reproduction instructions

1. Create a new block plugin with `npx @wordpress/create-block todo-list`
2. Run `npm run start`
3. Add a `render.php` file in the `src` folder
4. Add `"render": "file:./render.php"` to your `block.json` file.
5. See that the `render.php` never makes it to the `build` folder
6. Kill the script in your terminal
7. Run `npm run start` again.
8. See that the `render.php` now appears in your `build` folder

## block.json
```json
{
	"$schema": "https://schemas.wp.org/trunk/block.json",
	"apiVersion": 2,
	"name": "create-block/todo-list",
	"version": "0.1.0",
	"title": "Todo List",
	"category": "widgets",
	"icon": "smiley",
	"description": "Example block scaffolded with Create Block tool.",
	"supports": {
		"html": false
	},
	"textdomain": "todo-list",
	"editorScript": "file:./index.js",
	"editorStyle": "file:./index.css",
	"style": "file:./style-index.css",
	"render": "file:./render.php"
}
```

## render.php

```php
<?php
echo 'Hello World!';
```
## Screenshot of the src folder

<img width="284" alt="shows the src folder with a block.json file, edit.js file, editor.scss file, index.js file, render.php file, save.js file, and style.scss file" src="https://user-images.githubusercontent.com/6925260/231720831-e8be769f-75d0-4671-a69c-46bbdae76db3.png">

### Environment info

- WordPress 6.2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by gziolo** [comment]
Thank you for opening this issue, @aurooba. I haven’t tested it yet but I’m quite sure that the issue exists based on the following findings in the code.

There is a single place in the webpack config that collects all paths for all  `render` occurrences in existing `block.json` files:

https://github.com/WordPress/gutenberg/blob/c475dcb3ffee7627a8e1d2f13f364a7a9b195a61/packages/scripts/config/webpack.config.js#L42

It never gets recalculated in the watch mode, so it doesn't have a chance to detect a new file when processing inside the copy plugin:

https://github.com/WordPress/gutenberg/blob/c475dcb3ffee7627a8e1d2f13f364a7a9b195a61/packages/scripts/config/webpack.config.js#L273L283

**Comment by aurooba** [comment]
@gziolo That makes sense. I can take a look at this later this week and help fix it. I've been diving into the wp-scripts code to understand it better overall, so I've started to get a good handle on it.

**Comment by gziolo** [comment]
> I can take a look at this later this week and help fix it. I've been diving into the wp-scripts code to understand it better overall, so I've started to get a good handle on it.

Fantastic! 😀

**Comment by gziolo** [comment]
Fixed with  #50939.

**Comment by gziolo** [comment]
I opened a follow-up https://github.com/WordPress/gutenberg/pull/51162 with some performance optimization to the webpack configuration for detecting `render` PHP files.
